src/bees.cc

```diff
@@ -513,11 +515,13 @@ BeesTempFile::create()
 	m_ctx->insert_root_ino(m_fd);
 
 	// Set compression attribute
-	BEESTRACE("Getting FS_COMPR_FL on m_fd " << name_fd(m_fd));
-	int flags = ioctl_iflags_get(m_fd);
-	flags |= FS_COMPR_FL;
-	BEESTRACE("Setting FS_COMPR_FL on m_fd " << name_fd(m_fd) << " flags " << to_hex(flags));
-	ioctl_iflags_set(m_fd, flags);
+	if(!bees_tmpfile_compression_disabled) {
+		BEESTRACE("Getting FS_COMPR_FL on m_fd " << name_fd(m_fd));
+		int flags = ioctl_iflags_get(m_fd);
+		flags |= FS_COMPR_FL;
+		BEESTRACE("Setting FS_COMPR_FL on m_fd " << name_fd(m_fd) << " flags " << to_hex(flags));
+		ioctl_iflags_set(m_fd, flags);
+	}
```

Comment:
This logic is wrong as it won't prevent compression if the parent directory of the tmpfile has the compression attribute set. This would leave you with compression still active even when you said `no-tmpfile-compression` on the cmdline. That would be quite a surprising effect. Actually, I think it's better to get the updated kernel or force compression off for the whole filesystem if this kernel issue is a concern.